isis/src/base/objs/StretchBlob/StretchBlob.cpp

```diff
@@ -36,7 +36,16 @@ namespace Isis {
    *
    * @param name Name to use for Stretch
    */
-  StretchBlob::StretchBlob(QString name) : Blob(name, "Stretch"), m_stretch(name) {
+  StretchBlob::StretchBlob(QString name) {
+    m_stretch = CubeStretch(name, "Stretch");
+  }
+
+  StretchBlob::StretchBlob(Blob blob) {
+    stringstream os;
+    char *buff = blob.getBuffer();
+    std::string stringFromBuffer(buff);
+    m_stretch = CubeStretch(blob.Name(), blob.Type());
```

Comment:
I think this might be the wrong name and type for the CubeStretch? The blob's name is "CubeStretch" and its type is "Stretch" but the CubeStretch's name and type should be different. In the old version of the code, that information is stored in the Blob PVL as the name, stretch type, and band number. I'm confused about where this information is being stored in the updated code, but I'm not that familiar with the details of this project.